enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,343 @@
+   - Creates ignition configs by base64 encoding all YAML files from `manifests/` directory
+
+#### Step 2: Bootstrap Phase
+
+1. Bootstrap MCO recognizes MachineOSConfig manifests and stores them for later API creation
+2. Bootstrap MCO creates rendered MachineConfigs for each pool
+3. Bootstrap MCO then does the following to inject the pre-built container image into the rendered MC
+    - Maps MachineOSConfigs with pre-built annotations to their target pools
+    - Validates pre-built image format (digest format required)
+    - Sets `OSImageURL = preBuiltImage` in rendered MachineConfigs for pools with pre-built images and MachineOSConfigs
+    - Adds tracking annotation to modified MachineConfigs
+4. Bootstrap MCO creates ignition files with modified rendered MCs containing `OSImageURL` pointing to the pre-built container image
+5. Bootstrap MCO writes MachineOSConfig manifests to `/etc/mcs/bootstrap/machine-os-configs/` for post-bootstrap API creation
+6. The MCD first boot service detects the different `OSImageURL` value and uses `rpm-ostree rebase` to rebase the node to the custom pre-built container image
+7. After the cluster bootstrap completes, MCO starts and begins post-bootstrap processing
+
+#### Step 3: Post-Bootstrap Seeding Phase
+
+1. MCO controller's syncMachineOSConfigs() function reads MachineOSConfig files from `/etc/mcs/bootstrap/machine-os-configs/`
+2. MCO creates MachineOSConfig objects in the Kubernetes API if they don't already exist
+3. Build controller detects the `machineconfiguration.openshift.io/pre-built-image` annotation on the newly created MachineOSConfig
+4. Build controller triggers seeding workflow by calling `seedMachineOSConfigWithExistingImage()``
+5. MCO creates a synthetic MachineOSBuild with:
+    - Successful status referencing the seeded pre-built OS container image
+    - `PreBuiltImage` label set to "true"
+    - Proper metadata including MOSC name, DigestedImagePushSpec, and MCP reference
+    - Success condition with reason "PreBuiltImageSeeded"
+6. MCO updates MachineOSConfig status with `CurrentImagePullSpec` pointing to the pre-built image
+    - This creates consistency between the actual node state (already running pre-built image) and the MCO API objects for future OCL workflows
+
+#### Step 4: Post-Install Integration
+
+1. Any future updates/changes will use the current post-install workflow to trigger a new image build and roll it out onto the nodes based on the MachineOSConfig created at install time
+
+#### Notes and Restrictions:
+
+- A pre-built custom OS container image is needed. Image build during bootstrap is not supported
+- The Containerfile used to build the image needs to be added to the MachineOSConfig CRs created at install time
+- Two boots will occur: Initial disk image boot with stock RHCOS image and a second boot when the MCD first boot service runs to boot into the custom OS container image
```

Comment:
To reduce confusion, this is the same as we have today without OCL, so maybe we can just remove this part?